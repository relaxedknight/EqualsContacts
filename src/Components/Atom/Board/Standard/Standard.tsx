import type { FunctionComponent, ReactNode } from 'react'
import { cloneElement } from 'react'

import * as style from './Standard.css'
import * as Type from './Standard.type'

export const Standard: FunctionComponent<Type.Prop> = (prop) => {

  const mutated = cloneElement(prop.content, {
    className: style.content
  })

  return <section 
    className={style.container} 
    data-testid={prop.testId}>
    <header className={style.header}>{prop.header}</header>
    {mutated}
    <footer className={style.footer}>{prop.footer}</footer>
  </section>
}
