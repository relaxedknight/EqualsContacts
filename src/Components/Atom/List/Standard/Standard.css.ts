import { style } from '@vanilla-extract/css'

import { variable } from '@style/reset.css'

export const container = style({
  listStyle: 'none'
})

export const item = style({
  'selectors': {
    [`${container} > &`]: {
      marginBottom: variable.spacing.m
    },
    [`${container} > &:last-child`]: {
      marginBottom: 0
    }
  }
})
