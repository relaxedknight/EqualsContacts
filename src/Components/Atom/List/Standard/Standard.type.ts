import type { ReactNode } from 'react'

export type Prop = {
  className?: string
  items: ReactNode[]
  type: 'ul' | 'ol'
}
