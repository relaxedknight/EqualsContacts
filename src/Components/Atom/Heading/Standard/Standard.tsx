import type { FunctionComponent, ReactNode } from 'react'

import { className } from '@library'

import * as style from './Standard.css'

export const Standard: FunctionComponent<{
  className?: string
  children?: ReactNode
  type: 'h1' | 'h2' | 'h3' | 'h4' | 'h5'
}> = (prop) => {

  const Tag = prop.type

  return <Tag className={className.format(style.heading[prop.type], prop.className)}>{prop.children}</Tag>
}
