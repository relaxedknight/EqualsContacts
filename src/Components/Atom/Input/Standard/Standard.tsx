import { Atom } from '@Components'
import { className } from '@library'

import * as style from './Standard.css'
import * as Type from './Standard.type'

export function Standard<F extends Record<string, unknown>>(prop: Type.Prop<F>) {

  const registration = prop.register?.(prop.name, prop.validation)
  
  return <div 
    className={className.format(style.container, prop.className?.container)} 
    data-testid={prop.testId}>

    <label 
      className={className.format(style.label, prop.className?.label)}
      htmlFor={prop.id}>{prop.label}</label>

    <input 
      className={className.format(style.input, prop.className?.input)}
      id={prop.id}
      placeholder={prop.placeholder}
      required={prop.required}
      type={prop.type}
      {...registration}
      onChange={(e) => {

        prop.onChange?.(e.currentTarget.value)
        registration?.onChange(e)
      }} />

    {prop.error && <Atom.Text.Error>{prop.error}</Atom.Text.Error>}
  </div>
}
