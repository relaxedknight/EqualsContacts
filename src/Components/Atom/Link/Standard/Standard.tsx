import type { FunctionComponent } from 'react'

import * as style from './Standard.css'
import * as Type from './Standard.type'

export const Standard: FunctionComponent<Type.Prop> = (prop) => {

  const href = /^\d{3}-\d{3}-\d{4}$/.test(prop.href) ? `tel:${prop.href}` : prop.href

  return <a 
    className={style.text} 
    href={href}
    title={prop.title}>{prop.children}</a>
}
