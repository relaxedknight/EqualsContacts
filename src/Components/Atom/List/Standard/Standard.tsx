import type { FunctionComponent, ReactNode } from 'react'

import { className } from '@library'

import * as style from './Standard.css'
import * as Type from './Standard.type'

export const Standard: FunctionComponent<Type.Prop> = (prop) => {

  const List = 'ul'

  return <List className={className.format(style.container, prop.className)}>
    {prop.items.map((item, i) => {

      return <li
        className={style.item} 
        key={i}>{item}</li>
    })}
  </List>
}
