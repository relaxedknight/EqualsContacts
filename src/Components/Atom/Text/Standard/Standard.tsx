import type { FunctionComponent } from 'react'

import { className } from '@library'

import * as style from './Standard.css'
import * as Type from './Standard.type'

export const Standard: FunctionComponent<Type.Prop> = ({
  type: Type = 'p',
  ...prop
}) => {

  return <Type className={className.format(style.text, prop.className)}>{prop.children}</Type>
}
