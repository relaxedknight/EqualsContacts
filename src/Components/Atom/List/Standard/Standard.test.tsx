import { render } from '@testing-library/react'

import { Standard } from './Standard'
import * as style from './Standard.css'
import * as Type from './Standard.type'

const prop: Type.Prop = {
  items: [
    'Item 1',
    'Item 2',
    'Item 3'
  ],
  type: 'ul'
}

describe('Component/Atoms/List/Ordered', () => {

  it('renders with the correct attributes', () => {

    const screen = render(<Standard {...prop} />)
    const list = screen.getByRole('list')

    const expected = {
      className: style.container
    }

    Object.entries(expected).forEach((expected) => {
      expect(list).toHaveProperty(...expected)
    })
  })

  it.each<Type.Prop['type']>(['ul', 'ol'])('renders as %s when passed', (type) => {

    const screen = render(<Standard 
      {...prop}
      type={type} 
    />)
    const list = screen.getByRole('list')
  })

  describe('items', () => {

    const Element = 'div'

    it.each`
      type              | items
      ${'strings'}      | ${['1', '2', '3']}
      ${'JSX Elements'} | ${['1', '2', '3'].map((text) => <Element>{`Item ${text}`}</Element>)}
      ${'mixed'}        | ${['1', <Element>Item 2</Element>, '3']}
    `('renders when items are $type', ({ items, type }) => {

      const screen = render(<Standard 
        {...prop}
        items={items}
        type='ul'
      />)
      const list = screen.getByRole('list')

      expect(list).toMatchSnapshot()
    })
  })
})
