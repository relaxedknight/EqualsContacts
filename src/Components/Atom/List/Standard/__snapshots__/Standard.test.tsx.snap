// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/Atoms/List/Ordered items renders when items are JSX Elements 1`] = `
<ul
  class="Standard_container__1w14wn10"
  data-testid="Standard"
>
  <li
    class="Standard_item__1w14wn11"
  >
    <div>
      Item 1
    </div>
  </li>
  <li
    class="Standard_item__1w14wn11"
  >
    <div>
      Item 2
    </div>
  </li>
  <li
    class="Standard_item__1w14wn11"
  >
    <div>
      Item 3
    </div>
  </li>
</ul>
`;

exports[`Component/Atoms/List/Ordered items renders when items are mixed 1`] = `
<ul
  class="Standard_container__1w14wn10"
  data-testid="Standard"
>
  <li
    class="Standard_item__1w14wn11"
  >
    1
  </li>
  <li
    class="Standard_item__1w14wn11"
  >
    <div>
      Item 2
    </div>
  </li>
  <li
    class="Standard_item__1w14wn11"
  >
    3
  </li>
</ul>
`;

exports[`Component/Atoms/List/Ordered items renders when items are strings 1`] = `
<ul
  class="Standard_container__1w14wn10"
  data-testid="Standard"
>
  <li
    class="Standard_item__1w14wn11"
  >
    1
  </li>
  <li
    class="Standard_item__1w14wn11"
  >
    2
  </li>
  <li
    class="Standard_item__1w14wn11"
  >
    3
  </li>
</ul>
`;
